CodeGeneration/Sources/generate-swift-syntax/templates/swiftsyntax/SyntaxEnumFile.swift

```diff
@@ -57,4 +57,47 @@ let syntaxEnumFile = SourceFileSyntax(leadingTrivia: copyrightHeader) {
       }
     }
   }
+
+  for base: SyntaxNodeKind in [.decl, .stmt, .expr, .type, .pattern] {
+    let baseName = base.rawValue.withFirstCharacterUppercased;
+    try! EnumDeclSyntax(
+      """
+      /// Enum to exhaustively switch over all different \(raw: baseName) syntax nodes.
+      public enum \(raw: baseName)SyntaxEnum
```

Comment:
What do you think about adding the name creation directly to the `SyntaxNodeKind`? This would centralize the source for all types generated from `SyntaxNodeKind`, ensuring consistency and reducing manual repetition if one would like to use the name in different part of code generation package. The code could be refactored as follows:

```swift
public var baseTypeEnum: TypeSyntax {
  if isBase {
    return "\(syntaxType)Enum"
  } else {
    fatalError("Only base kinds can define base enum types.")
  }
}
```

Reply:
If anyone finds the actual use-case of it, they can propose it later.
Personally, I don't think a good idea to make it a method/property (unless it's used in _many_ places) because `enum \(raw: baseName)SyntaEnum` is clearer than `enum \(raw: base.baseTypeEnum)`.

Reply:
I see where you're coming from. However, considering that the name is used in multiple places within this code, I propose at least moving the name to a local variable. For instance:

```swift
let enumName: TypeSyntax = "\(syntaxType)Enum"
```